pkg/utils/k8sutil.go

```diff
@@ -372,6 +372,17 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		addInitContainerAnnotation(dpm)
 	}
 
+	// update deployment for memory request
+	if funcObj.Spec.Memory != "" {
+		//mem value has already been validated before
+		quantity, _ := resource.ParseQuantity(funcObj.Spec.Memory)
```

Comment:
.. just because it might have been parsed in kubeless client doesn't mean that the value you receive here came from a kubeless client.  We want users to also create Function TPRs directly, and there's always malicious/version-skewed clients, etc.